contracts/test/TestTokenMigration.sol

```diff
@@ -0,0 +1,323 @@
+
+    address preDeterministicJBDirectory = addressFrom(address(this), 15);
+
+    // JBFundingCycleStore
+    JBFundingCycleStore _jbFundingCycleStore = new JBFundingCycleStore(IJBDirectory(preDeterministicJBDirectory));
+
+    // JBDirectory
+    _v3jJbDirectory = new JBDirectory(jbOperatorStore(), _jbProjects, _jbFundingCycleStore, multisig());
+
+    // JBTokenStore
+    _v3JbTokenStore = new JBTokenStore(jbOperatorStore(), _jbProjects, _v3jJbDirectory);
+
+    // JBSplitsStore
+    JBSplitsStore _newJbSplitsStore = new JBSplitsStore(jbOperatorStore(), _jbProjects, _v3jJbDirectory);
+
+    // JBController
+    v3controller = new JBController(
+      jbOperatorStore(),
+      _jbProjects,
+      _v3jJbDirectory,
+      _jbFundingCycleStore,
+      _v3JbTokenStore,
+      _newJbSplitsStore
+    );
+
+    evm.prank(multisig());
+    _v3jJbDirectory.setIsAllowedToSetFirstController(address(v3controller), true);
+
+    // JBETHPaymentTerminalStore
+    JBSingleTokenPaymentTerminalStore _jbPaymentTerminalStore = new JBSingleTokenPaymentTerminalStore(
+      _v3jJbDirectory,
+      _jbFundingCycleStore,
+      jbPrices()
+    );
+
+    // AccessJBLib
+    AccessJBLib _accessJBLib = new AccessJBLib();
+
+    // JBETHPaymentTerminal
+    JBETHPaymentTerminal _jbETHPaymentTerminal = new JBETHPaymentTerminal(
+      _accessJBLib.ETH(),
+      jbOperatorStore(),
+      _jbProjects,
+      _v3jJbDirectory,
+      _newJbSplitsStore,
+      jbPrices(),
+      _jbPaymentTerminalStore,
+      multisig()
+    );
+
+   IJBPaymentTerminal[] memory _newProjectterminals = new IJBPaymentTerminal[](1);
+    _newProjectterminals[0] = (_jbETHPaymentTerminal);
+
+    _fundAccessConstraint.push(JBFundAccessConstraints({
+        terminal: _jbETHPaymentTerminal,
+        token: jbLibraries().ETHToken(),
+        distributionLimit: 0,
+        overflowAllowance: 0,
+        distributionLimitCurrency: jbLibraries().ETH(),
+        overflowAllowanceCurrency: jbLibraries().ETH()
+    }));
+
+   
+    _metadata.allowMinting = true;
+
+    // mimicing a v3 project launch since there was dependency issue with importing both v2 & v3 contract togethet since the almost every contract is the same
+    evm.prank(_projectOwner);
+    _projectId = v3controller.launchProjectFor(
+      _projectOwner,
+      _projectMetadata,
+      _data,
+      _metadata,
+      block.timestamp,
+      _groupedSplits,
+      _fundAccessConstraint,
+      _newProjectterminals,
+      ''
+    );
+
+    assertEq(_jbProjects.ownerOf(_projectId), _projectOwner);
+
+    TestBaseWorkflowV1.setUp();
+    
+    // deploying v3 token
+    _v3Token = new JBV3Token('v3 token', 'v3 token', _projectId, _v3jJbDirectory, ticketBooth(), jbTokenStore());
+
+  }
+
+  function testMigrationWhenUseHasClaimedAndUnclaimedBalances() public {
+    TerminalV1_1 _v1Terminal = terminal();
+    JBETHPaymentTerminal _v2Terminal = jbETHPaymentTerminal();
+
+    address _user = address(bytes20(keccak256('user')));
```

Comment:
makeadr() is now part of forge-std ;)

Reply:
yh i ran forge update lib/forge-std but for some reason it didn't update looks like some issue on my local so leaving this as it is for now